File: JSPM.Web/Modules/Examinations/ExamSections/ExamSectionsForm.cs
using Serenity.ComponentModel;
using System;

namespace JSPM.Examinations.Forms;

[FormScript("Examinations.ExamSections")]
[BasedOnRow(typeof(ExamSectionsRow), CheckNames = true)]
public class ExamSectionsForm
{
    public string Name { get; set; }
    public string DisplayName { get; set; }
    public Guid ExamId { get; set; }
    public string Instructions { get; set; }
    public int SectionDurationInSeconds { get; set; }
    public int NumberOfQuestions { get; set; }
    public int NumberofMandatoryQuestions { get; set; }
}

File: JSPM.Web/Modules/ServerTypes/Examination/ExamSectionsForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ExamSectionsForm {
    Name: StringEditor;
    DisplayName: StringEditor;
    ExamId: StringEditor;
    Instructions: StringEditor;
    SectionDurationInSeconds: IntegerEditor;
    NumberOfQuestions: IntegerEditor;
    NumberofMandatoryQuestions: IntegerEditor;
}

export class ExamSectionsForm extends PrefixedContext {
    static formKey = 'Examination.ExamSections';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamSectionsForm.init)  {
            ExamSectionsForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(ExamSectionsForm, [
                'Name', w0,
                'DisplayName', w0,
                'ExamId', w0,
                'Instructions', w0,
                'SectionDurationInSeconds', w1,
                'NumberOfQuestions', w1,
                'NumberofMandatoryQuestions', w1
            ]);
        }
    }
}